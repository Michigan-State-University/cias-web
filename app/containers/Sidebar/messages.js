/*
 * Sidebar Messages
 *
 * This contains all the text for the Sidebar container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.Sidebar';

export default defineMessages({
  sidebarNavigationHeader: {
    id: `${scope}.sidebarNavigationHeader`,
    defaultMessage: 'CIAS Panel',
  },
  showSidebarButtonLabel: {
    id: `${scope}.showSidebarButtonLabel`,
    defaultMessage: 'Hover to show sidebar',
  },
  showSidebarIcon: {
    id: `${scope}.showSidebarIcon`,
    defaultMessage: 'Show sidebar icon',
  },
});
