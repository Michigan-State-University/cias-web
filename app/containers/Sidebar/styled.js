import styled from 'styled-components';
import { Link } from 'react-router-dom';
import Color from 'color';

import { borders, boxShadows, elements, themeColors, ZIndex } from 'theme';
import { decimalToHex } from 'utils/hexUtils';

import { margin } from 'components/BaseComponentStyles';
import { ShowHiddenContentButton } from 'components/ShowHiddenContentButton';

export const SidebarStyled = styled.div`
  width: ${elements.sidebarWidth}px;
  top: 0;
  background-color: white;
  padding: 20px;
  box-shadow: ${boxShadows.black};
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  z-index: ${ZIndex.SIDEBAR};
  margin-top: ${elements.navbarHeight}px;
  height: calc(100vh - ${elements.navbarHeight}px);
  transition: left 0.4s ease;

  @media only screen and (max-width: 1280px) {
    position: fixed;
    left: calc(0px - ${elements.sidebarWidth}px);
    ${({ isVisible }) => (isVisible ? 'left: 0;' : '')}
  }
`;

export const SidebarItemLink = styled(Link)`
  display: flex;
  text-decoration: none;
  padding: 8px;
  width: max-content;

  ${({ $isActive }) =>
    $isActive
      ? {
          backgroundColor: `${themeColors.secondary}${decimalToHex(20)}`,
          borderRadius: borders.borderRadius,
          '*': { color: themeColors.secondary },
        }
      : {}}

  ${margin};
`;

export const SidebarSubItemsContainer = styled.div`
  padding: 12px 20px 12px 40px;
  display: flex;
  flex-direction: column;
  gap: 12px;
`;

export const SidebarSubItemLink = styled(Link)`
  text-decoration: none;
  font-size: 15px;
  line-height: 150%;
  ${({ $isActive }) =>
    $isActive
      ? {
          fontWeight: 700,
          color: themeColors.text,
        }
      : {
          fontWeight: 400,
          color: Color(themeColors.text).alpha(0.7),
        }}
`;

export const ShowSidebarButton = styled(ShowHiddenContentButton)`
  position: absolute;
  top: 16px;
  right: -40px;
  box-shadow: 20px 0 20px rgba(0, 0, 0, 0.08);
  @media only screen and (min-width: 1400px) {
    display: none;
  }
`;
