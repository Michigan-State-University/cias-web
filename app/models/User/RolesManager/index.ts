export * from './defaultNavbarTabs';
export * from './useRoleManager';
export * from './UserRoles';
export * from './RolesColor';
export * from './navbarNames';
