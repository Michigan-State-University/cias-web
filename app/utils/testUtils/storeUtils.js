import { createStore } from 'redux';
import createReducer from 'reducers';
import { initialState as authReducerInitialState } from 'global/reducers/auth/reducer';

export const createTestStore = (initialState) => {
  const reducer = (state) => state;

  const store = createStore(reducer, {
    auth: authReducerInitialState,
    ...initialState,
  });

  store.runSaga = () => {};
  store.injectedReducers = {};
  store.injectedSagas = {};
  store.createReducer = createReducer;

  return store;
};
