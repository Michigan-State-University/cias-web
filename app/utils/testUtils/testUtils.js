import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { DEFAULT_LOCALE } from 'i18n';

const ProviderWrapper = ({ children }) => (
  <IntlProvider locale={DEFAULT_LOCALE}>{children}</IntlProvider>
);

const testRender = (ui, options) =>
  render(ui, { wrapper: ProviderWrapper, ...options });

export { testRender };

ProviderWrapper.propTypes = {
  children: PropTypes.node,
};
