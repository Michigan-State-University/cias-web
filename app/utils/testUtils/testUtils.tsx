import React, { ComponentType, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { DEFAULT_LOCALE } from 'i18n';

type Props = { children: JSX.Element };

const ProviderWrapper = ({ children }: Props): JSX.Element => (
  <IntlProvider locale={DEFAULT_LOCALE}>{children}</IntlProvider>
);

const testRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult =>
  render(ui, { wrapper: ProviderWrapper as ComponentType, ...options });

export { testRender };
