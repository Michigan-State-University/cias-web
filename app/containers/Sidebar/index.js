/**
 *
 * Sidebar
 *
 */

import React, { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { injectIntl, useIntl } from 'react-intl';
import { compose } from 'redux';

import { makeSelectUser } from 'global/reducers/auth';
import { NAVIGATION } from 'utils/navbarNames';

import Comment from 'components/Text/Comment';

import DefaultSidebar from './components/DefaultSidebar';

import { SidebarStyled } from './styled';
import messages from './messages';

export function Sidebar({ user: { roles }, sidebarProps }) {
  const { formatMessage } = useIntl();

  const { activeTab, sidebarId } = sidebarProps || {};

  const renderSidebar = useCallback(() => {
    if (sidebarId === NAVIGATION.DEFAULT)
      return <DefaultSidebar activeTab={activeTab} userRole={roles[0]} />;

    return null;
  }, [sidebarId, activeTab, roles]);

  return (
    <SidebarStyled>
      <Comment mb="8px !important">
        {formatMessage(messages.sidebarNavigationHeader)}
      </Comment>
      {renderSidebar()}
    </SidebarStyled>
  );
}

Sidebar.propTypes = {
  user: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    roles: PropTypes.arrayOf(PropTypes.string),
    avatar: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  }),
  sidebarProps: PropTypes.shape({
    sidebarId: PropTypes.string.isRequired,
    activeTab: PropTypes.string,
  }),
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
});

export default memo(
  compose(
    connect(mapStateToProps),
    injectIntl,
  )(Sidebar),
);
