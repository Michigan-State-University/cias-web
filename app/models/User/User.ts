import { CamelToSnakeOmitId } from 'global/types/camelToSnake';
import { Roles } from './RolesManager/UserRoles';

export interface User {
  active: boolean;
  email: string;
  emailNotification: boolean;
  feedbackCompleted: boolean;
  firstName: string;
  fullName: string;
  id: string;
  lastName: string;
  phoneName: Nullable<string>;
  roles: Roles[];
  smsNotification: boolean;
  teamId: Nullable<string>;
  teamName: Nullable<string>;
}

export type UserDTO = CamelToSnakeOmitId<User>;
