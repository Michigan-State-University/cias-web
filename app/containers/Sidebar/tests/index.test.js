/**
 *
 * Tests for Sidebar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import 'jest-styled-components';

import { DEFAULT_LOCALE } from 'i18n';
import { Roles } from 'models/User/RolesManager';
import { createTestStore } from 'utils/testUtils/storeUtils';

import Sidebar from '../index';

describe('<Sidebar />', () => {
  let store;
  const initialState = {
    auth: {
      user: { firstName: 'test', lastName: 'test', roles: [Roles.Admin] },
    },
    session: {
      session: {
        id: '12dasc0123=21-2',
        name: 'e-Session New',
      },
    },
  };

  beforeAll(() => {
    store = createTestStore(initialState);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Sidebar logOut={() => {}} path="/interventions" />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot with session path', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Sidebar logOut={() => {}} path="/interventions" />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
  it('Should render and match the snapshot without matching path', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <MemoryRouter>
          <IntlProvider locale={DEFAULT_LOCALE}>
            <Sidebar logOut={() => {}} path="" />
          </IntlProvider>
        </MemoryRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
