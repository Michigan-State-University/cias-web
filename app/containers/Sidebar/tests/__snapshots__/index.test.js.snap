// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Should render and match the snapshot with session path 1`] = `
.c4 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 8px !important;
  color: #2F3850;
}

.c5 {
  margin: 0;
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  color: #6B707C;
  margin-bottom: 8px !important;
}

.c3 {
  -webkit-transition: -webkit-transform 0.4s ease;
  -webkit-transition: transform 0.4s ease;
  transition: transform 0.4s ease;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  border-width: 0;
  background-color: #FFFFFF;
  border-radius: 0 5px 5px 0;
  width: 40px;
  height: 40px;
}

.c0 {
  width: 280px;
  top: 0;
  background-color: white;
  padding: 20px;
  box-shadow: 0px 0px 50px rgba(0,0,0,0.08);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 10;
  margin-top: 70px;
  height: calc(100vh - 70px);
  -webkit-transition: left 0.4s ease;
  transition: left 0.4s ease;
}

.c2 {
  position: absolute;
  top: 16px;
  right: -40px;
  box-shadow: 20px 0 20px rgba(0,0,0,0.08);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
  width: 30px;
  height: 30px;
}

.c11:after {
  content: ' ';
  display: block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 6px solid #B727EA;
  border-color: #B727EA transparent #B727EA transparent;
  -webkit-animation: hAjVGa 1.2s linear infinite;
  animation: hAjVGa 1.2s linear infinite;
}

.c9 {
  width: 100%;
  height: 40px;
  border-radius: 5px;
  cursor: pointer;
  border: none;
  outline: none;
  font-weight: bold;
  background-color: #FFFFFF;
  color: #B727EA;
  border: 1px solid #B727EA;
  -webkit-transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  margin-top: 20px;
  margin-bottom: 20px;
  width: auto;
  padding-right: 10px;
  padding-left: 10px;
}

.c9:hover {
  background-color: #B727EA;
  color: #FFFFFF;
  border: 1px solid transparent;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (max-width:1280px) {
  .c0 {
    position: fixed;
    left: calc(0px - 280px);
  }
}

@media only screen and (min-width:1400px) {
  .c2 {
    display: none;
  }
}

<div
  class="c0"
>
  <button
    aria-label="Hover to show sidebar"
    class="c1 c2"
    title="Hover to show sidebar"
  >
    <img
      alt="Show sidebar icon"
      class="c3"
      src="IMAGE_MOCK"
    />
  </button>
  <p
    class="c4 c5"
  >
    CIAS Panel
  </p>
</div>
`;

exports[`<Sidebar /> Should render and match the snapshot without matching path 1`] = `
.c4 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-bottom: 8px !important;
  color: #2F3850;
}

.c5 {
  margin: 0;
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  color: #6B707C;
  margin-bottom: 8px !important;
}

.c3 {
  -webkit-transition: -webkit-transform 0.4s ease;
  -webkit-transition: transform 0.4s ease;
  transition: transform 0.4s ease;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  border-width: 0;
  background-color: #FFFFFF;
  border-radius: 0 5px 5px 0;
  width: 40px;
  height: 40px;
}

.c0 {
  width: 280px;
  top: 0;
  background-color: white;
  padding: 20px;
  box-shadow: 0px 0px 50px rgba(0,0,0,0.08);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  z-index: 10;
  margin-top: 70px;
  height: calc(100vh - 70px);
  -webkit-transition: left 0.4s ease;
  transition: left 0.4s ease;
}

.c2 {
  position: absolute;
  top: 16px;
  right: -40px;
  box-shadow: 20px 0 20px rgba(0,0,0,0.08);
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: auto;
  width: 30px;
  height: 30px;
}

.c11:after {
  content: ' ';
  display: block;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  border: 6px solid #B727EA;
  border-color: #B727EA transparent #B727EA transparent;
  -webkit-animation: hAjVGa 1.2s linear infinite;
  animation: hAjVGa 1.2s linear infinite;
}

.c9 {
  width: 100%;
  height: 40px;
  border-radius: 5px;
  cursor: pointer;
  border: none;
  outline: none;
  font-weight: bold;
  background-color: #FFFFFF;
  color: #B727EA;
  border: 1px solid #B727EA;
  -webkit-transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  transition: background-color 300ms ease,color 300ms ease,border 300ms ease;
  margin-top: 20px;
  margin-bottom: 20px;
  width: auto;
  padding-right: 10px;
  padding-left: 10px;
}

.c9:hover {
  background-color: #B727EA;
  color: #FFFFFF;
  border: 1px solid transparent;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (max-width:1280px) {
  .c0 {
    position: fixed;
    left: calc(0px - 280px);
  }
}

@media only screen and (min-width:1400px) {
  .c2 {
    display: none;
  }
}

<div
  class="c0"
>
  <button
    aria-label="Hover to show sidebar"
    class="c1 c2"
    title="Hover to show sidebar"
  >
    <img
      alt="Show sidebar icon"
      class="c3"
      src="IMAGE_MOCK"
    />
  </button>
  <p
    class="c4 c5"
  >
    CIAS Panel
  </p>
</div>
`;
